src/nomad_material_processing/__init__.py

```diff
@@ -123,6 +244,154 @@ class Parallelepiped(Geometry):
+    )
+    lower_cap_surface_area = Quantity(
+        type=float,
+        description='Area of the lower cap.',
+        a_eln=ELNAnnotation(
+            component='NumberEditQuantity',
+            defaultDisplayUnit='millimeter ** 2',
+        ),
+        unit='meter ** 2',
+    )
+    upper_cap_surface_area = Quantity(
+        type=float,
+        description='Area of the upper cap.',
+        a_eln=ELNAnnotation(
+            component='NumberEditQuantity',
+            defaultDisplayUnit='millimeter ** 2',
+        ),
+        unit='meter ** 2',
+    )
+    lateral_surface_area = Quantity(
+        type=float,
+        description='Area of the lateral surface.',
+        a_eln=ELNAnnotation(
+            component='NumberEditQuantity',
+            defaultDisplayUnit='millimeter ** 2',
+        ),
+        unit='meter ** 2',
+    )
+
+
+class Cylinder(Geometry):
+    """
+    Class autogenerated from yaml schema.
+    """
+
+    m_def = Section()
+    height = Quantity(
+        type=float,
+        description='The z dimension (or thickness) of the parallelepiped.',
+        a_eln=ELNAnnotation(
+            component='NumberEditQuantity',
+            defaultDisplayUnit='nanometer',
+        ),
+        label='Height (z or c)',
+        unit='meter',
+    )
+    radius = Quantity(
+        type=float,
+        description='Radius of the cylinder.',
+        a_eln=ELNAnnotation(
+            component='NumberEditQuantity',
+            defaultDisplayUnit='millimeter',
+        ),
+        unit='meter',
+    )
+    lower_cap_surface_area = Quantity(
+        type=float,
+        description='Area of the lower cap.',
+        a_eln=ELNAnnotation(
+            component='NumberEditQuantity',
+            defaultDisplayUnit='millimeter ** 2',
+        ),
+        unit='meter ** 2',
+    )
+    cap_surface_area = Quantity(
+        type=float,
+        description='Area of the cap.',
+        a_eln=ELNAnnotation(
+            component='NumberEditQuantity',
+            defaultDisplayUnit='millimeter ** 2',
+        ),
+        unit='meter ** 2',
+    )
+    lateral_surface_area = Quantity(
+        type=float,
+        description='Area of the lateral surface.',
+        a_eln=ELNAnnotation(
+            component='NumberEditQuantity',
+            defaultDisplayUnit='millimeter ** 2',
+        ),
+        unit='meter ** 2',
+    )
+
+
+class CylinderSector(Cylinder):
+    central_angle = Quantity(
+        type=float,
+        description="""The angle that defines the portion of the cylinder.
+        This angle is taken at the center of the base circle
+        and extends to the arc that defines the cylindrical sector.""",
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.NumberEditQuantity,
+        ),
+        unit='degree',
+    )
+
+
+class Irregular(Geometry):
+    """
+    A shape that does not fit into any of the other geometry classes.
+    """
+
+    m_def = Section()
+    height = Quantity(
```

Comment:
A height would imply that you have to parallel surfaces though. Maybe we name it something like `ParallelSurfaces`?